File: PYTHON/menstrualcyclecalculator.py
print("""
		Menstral cycle App
""")

last_cycle_start_date = int(input("Enter last cycle start date (in days, e.g., 1-31): "))

current_cycle_start_date = int(input("Enter current cycle start date (in days, e.g., 1-31): "))

if last_cycle_start_date > current_cycle_start_date:
    temp = last_cycle_start_date
    last_cycle_start_date = current_cycle_start_date
    current_cycle_start_date = temp

cycle_length = current_cycle_start_date - last_cycle_start_date

ovulation_day = last_cycle_start_date + cycle_length // 2

print(f"The length of your cycle is: {cycle_length}")

print(f"Your ovulation day is approximately on day: {ovulation_day}")

safe_period_start = ovulation_day + 2
safe_period_end = last_cycle_start_date + cycle_length

print(f"Your safe period is approximately from day {safe_period_start} to day {safe_period_end}")

File: JAVA/MenstrualCycleCalculator.java
import java.util.Scanner;
public class MenstrualCycleCalculator {
	public static void main(String[] args) {
	Scanner input = new Scanner(System.in);

	int Temp;
	int lastCycleStartDate;
	int currentCycleStartDate;
	int cycleLength;

	System.out.print("Enter last cycle start date (in days, e.g., 1-31): ");
	lastCycleStartDate = input.nextInt();

	System.out.print("Enter current cycle start date (in days, e.g., 1-31): ");
	currentCycleStartDate = input.nextInt();

	if(lastCycleStartDate > currentCycleStartDate){
		lastCycleStartDate = lastCycleStartDate;
		}
	else if (lastCycleStartDate < currentCycleStartDate){
		Temp = lastCycleStartDate;
		lastCycleStartDate = currentCycleStartDate;
		currentCycleStartDate = Temp;
		}


	cycleLength = currentCycleStartDate - lastCycleStartDate;

	int ovulationDay = lastCycleStartDate + cycleLength / 2;

	System.out.println("The length of your cycle is: " + cycleLength);

	System.out.println("Your ovulation day is approximately on day: " + ovulationDay);


	int safePeriodStart = ovulationDay + 2; 

	int safePeriodEnd = lastCycleStartDate + cycleLength; 

	System.out.println("Your safe period is approximately from day " + safePeriodStart + " to day " + safePeriodEnd);


}
}

